src/Tizen.Messaging/Tizen.Messaging.Messages/MessagesManagerImpl.cs

```diff
@@ -102,13 +102,18 @@ private void deinitialize()
 
         internal Task<SentResult> SendMessageAsync(Message message, bool saveToSentbox)
         {
-            var task = new TaskCompletionSource<SentResult>();
+            SentResult response;
+            TaskCompletionSource<SentResult> task = new TaskCompletionSource<SentResult>();
 
-            _messageSentCallback = (int result, IntPtr data) =>
+            lock (this)
             {
-                task.SetResult((SentResult)result);
+                _messageSentCallback = async (int result, IntPtr data) =>
+                {
+                    response = (SentResult)result;
+                    await task.Task.ConfigureAwait(false);
+                    task.SetResult(response);
```

Comment:
What is your intent?
It seems to deadlock, 
113 line - it wait completion of task (it wait execute of 114 line)
114 line - it make completion of task (it can execute 113 line was completed)